### PR TITLE
better MacOS support

### PR DESCRIPTION
**Pull request by cddigi**

### CODE REVIEW

**Comment by BBC-Esq** [comment]
Thanks for the commits.  Give me a little time to review them since it's the holidays, but I'm definitely all for any improvements for systems other than Windows.

File: src/metrics_bar.py
Comment (BBC-Esq) [inline src/metrics_bar.py:1]: This change was already made on my computer but apparently hadn't been uploaded to github...will do so shortly.

**Comment by BBC-Esq** [comment]
I uploaded a revised version of choose_documents.py, which I believe should address your concern about properly opening folders in Mac.  I'd already refactored to us Pathlib, so I simply added your concern and modified it slightly to also address Linux users.  This is the relevant portion:

```
    # Cross-platform directory opening
    if os.name == 'nt':  # Windows
        subprocess.Popen(f'explorer "{str(docs_folder)}"')
    elif os.name == 'posix':  # MacOS and Linux
        subprocess.Popen(f'xdg-open "{str(docs_folder)}"')
```
Thanks, let me know if there are other scripts that don't work perfectly with MacOS or LInux...

**Comment by BBC-Esq** [comment]
I believe I added (or they were already added just not uploaded) everything in the pull request.  If you could, please re-submit another pull request based on the most recent release's code, specifically, the .gitignore and pyproject.toml stuff...I've never used that before and it'd help me to have a single pull request to look at with just that.  Thanks!

**Comment by cddigi** [comment]
> ~written by mixtral-8x7b-instruct

Sure, I'd be happy to explain!

`.gitignore` is a file used by Git version control system to determine which files or directories should be ignored when committing code to the repository. By adding specific patterns to this file, you can prevent accidentally committing sensitive information, generated files, or other files that don't need to be tracked by version control.

For a Python project managed with Poetry, you might want to include patterns such as:

* `__pycache__/` - directory for compiled Python files
* `*.pyc` - compiled Python files
* `*.swp` - Vim swap files
* `*.env` - environment variables file
* `*.DS_Store` - macOS system file

Poetry uses a configuration file called `pyproject.toml` to manage the project's dependencies, build configuration, and metadata. Here are some of the common sections you might see in a `pyproject.toml` file:

* `[tool.poetry]` - Poetry configuration section, where you specify the project's name, version, description, authors, and other metadata
* `[tool.poetry.dependencies]` - section for specifying the project's dependencies, including the package name, version range, and optional features
* `[tool.poetry.dev-dependencies]` - section for specifying the development dependencies required to develop and test the project
* `[build-system]` - section required by PEP 517/518 build backend, specifying the minimum build system requirements

Here's an example `pyproject.toml` file for a Poetry-managed Python project:
```makefile
[tool.poetry]
name = "my-python-project"
version = "0.1.0"
description = "A Python project managed with Poetry"
authors = ["Your Name <your.email@example.com>"]
license = "MIT"

[tool.poetry.dependencies]
python = "^3.8"
numpy = "*"
pandas = ">=1.2.0"

[tool.poetry.dev-dependencies]
pytest = "*"
black = "*"
isort = "*"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```
In this example, the project depends on `numpy` and `pandas`, and requires Python 3.8 or higher to run. The development dependencies include `pytest`, `black`, and `isort` for testing, formatting, and sorting imports, respectively.

I hope that helps! Let me know if you have any further questions.

**Comment by BBC-Esq** [comment]
That helps, thank you.  Did you see my message about trying to separate out the different modifications into separate pull requests?  Not even sure if that's possible...

**Comment by cddigi** [comment]
Yeah, I was not aware of the mess that was to be had managing python package dependencies. Came across Poetry and seemed to be a good one. I've been learning python as I go with the aid of GPT-4.

**Pull request closed without merging**